sdk/cosmos/azure-cosmos-benchmark/src/main/java/com/azure/cosmos/benchmark/AsynReadWithMultipleClients.java

```diff
@@ -0,0 +1,299 @@
+        String splitBy = ",";
+        try {
+            br = new BufferedReader(new FileReader(csvFile));
+            while ((line = br.readLine()) != null) {
+                String[] hostAndKey = line.split(splitBy);
+                if (hostAndKey.length == 2) {
+                    CosmosAsyncClient asyncClient = new CosmosClientBuilder()
+                        .endpoint(hostAndKey[0])
+                        .key(hostAndKey[1])
+                        .connectionPolicy(configuration.getConnectionPolicy())
+                        .consistencyLevel(configuration.getConsistencyLevel())
+                        .connectionReuseAcrossClientsEnabled(true)
+                        .buildAsyncClient();
+                    List<PojoizedJson> docsToRead = new ArrayList<>();
+                    if (!configuration.isDeleteCollections()) {
+                        CosmosAsyncDatabase cosmosAsyncDatabase = asyncClient.createDatabaseIfNotExists(this.configuration.getDatabaseId()).block().getDatabase();
+                        CosmosAsyncContainer cosmosAsyncContainer = cosmosAsyncDatabase.createContainerIfNotExists(configuration.getCollectionId(), PARTITION_KEY, 100000).block().getContainer();
+                        String partitionKey = cosmosAsyncContainer.read().block().getProperties().getPartitionKeyDefinition()
+                            .getPaths().iterator().next().split("/")[1];
+                        String dataFieldValue = RandomStringUtils.randomAlphabetic(this.configuration.getDocumentDataFieldSize());
+                        ArrayList<Flux<PojoizedJson>> createDocumentObservables = new ArrayList<>();
+
+                        for (int i = 0; i < this.configuration.getNumberOfPreCreatedDocuments(); i++) {
+                            String uuid = UUID.randomUUID().toString();
+                            com.azure.cosmos.benchmark.PojoizedJson newDoc = generateDocument(uuid, dataFieldValue, partitionKey);
+
+                            Flux<PojoizedJson> obs = cosmosAsyncContainer.createItem(newDoc).map(resp -> {
+                                    com.azure.cosmos.benchmark.PojoizedJson x =
+                                        resp.getItem();
+                                    return x;
+                                }
+                            ).flux();
+                            createDocumentObservables.add(obs);
+                        }
+                        docsToRead = Flux.merge(Flux.fromIterable(createDocumentObservables), 100).collectList().block();
+                        MeterRegistry registry = configuration.getAzureMonitorMeterRegistry();
+
+                        if (registry != null) {
+                            BridgeInternal.monitorTelemetry(registry);
+                        }
+
+                        registry = configuration.getGraphiteMeterRegistry();
+
+                        if (registry != null) {
+                            BridgeInternal.monitorTelemetry(registry);
```

Comment:
meterRegistry is a static setting, why do we need to set it "inside" the for loop? shouldn't this be outside?

Reply:
done